BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -112,6 +125,34 @@ public virtual void Initialize(TelemetryConfiguration configuration, TelemetryMo
             this.Initialize(configuration, modules, PlatformSingleton.Current.ReadConfigurationXml());
         }
 
+        [SuppressMessage("Microsoft.Reliability", "CA2000:DisposeObjectsBeforeLosingScope", Justification = "We want objects created in this method to live for the life of the application.")]
+        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", Justification = "This value is not used in NetStandard1.3 but is used for all other frameworks..")]
+        internal void EvaluateSelfDiagnosticsMode(TelemetryModules modules)
+        {
+#if !NETSTANDARD1_3
+            if (PlatformSingleton.Current.TryGetEnvironmentVariable(SelfDiagnosticsEnvironmentVariable, out string selfDiagnosticsConfigurationString))
+            {
+                var keyValuePairs = SelfDiagnosticsProvider.ParseConfigurationString(selfDiagnosticsConfigurationString);
+
+                if (SelfDiagnosticsProvider.IsFileDiagnostics(keyValuePairs, out string path, out string level))
+                {
+                    var module = modules?.Modules.SingleOrDefault(x => x is FileDiagnosticsTelemetryModule);
+                    if (module != null)
+                    {
+                        modules.Modules.Remove(module);
+                    }
+
+                    modules.Modules.Add(new FileDiagnosticsTelemetryModule
```

Comment:
modules would be null for asp.net core, worker or wherever TelemetryConfiguration.CreateDefault() is used. And in all those cases, this diagnosticmodule will fail to be added.

(it works currently in asp.net core purely by luck - because it 'touches' TC.Active to maintain backward compatibility. But that's not a behavior this feature can count on, as we'll be deprecating .Active in next version change, which will break this diagnostic feature unintentionally)